engine/access/rest/routes/account_keys.go

```diff
@@ -0,0 +1,47 @@
+package routes
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/access"
+	"github.com/onflow/flow-go/engine/access/rest/models"
+	"github.com/onflow/flow-go/engine/access/rest/request"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// GetAccountKeyByID handler retrieves an account key by address and ID and returns the response
+func GetAccountKeyByID(r *request.Request, backend access.API, link models.LinkGenerator) (interface{}, error) {
+	req, err := r.GetAccountKeyRequest()
+	if err != nil {
+		return nil, models.NewBadRequestError(err)
+	}
+
+	header, _, err := backend.GetLatestBlockHeader(r.Context(), false)
+	if err != nil {
+		return nil, err
+	}
+
+	account, err := backend.GetAccountAtBlockHeight(r.Context(), req.Address, header.Height)
+	if err != nil {
+		return nil, err
+	}
+
+	var accountKey flow.AccountPublicKey
+	found := false
+	for _, key := range account.Keys {
+		if key.Index == int(req.KeyID) {
+			accountKey = key
```

Comment:
nit: you could just return the found key as a response which would remove the need for found and another nested if bellow. You could just return not found at the end.

Reply:
Nice suggestion :rocket: 
Updated in https://github.com/onflow/flow-go/pull/4583/commits/29402e349eb8e6b86a5da8f4e530c2a817e09dea